### PR TITLE
Remove hint dist sql

### LINKED ISSUE #23953: Remove Hint DistSQL
Hi community,

Since [SQL Hint](https://shardingsphere.apache.org/document/current/en/user-manual/shardingsphere-jdbc/special-api/sharding/hint/#use-special-sql-comments) is more convenient and compatible with Proxy, I think it is time to consider removing [Hint DistSQL](https://shardingsphere.apache.org/document/current/en/user-manual/shardingsphere-proxy/distsql/syntax/ral/hint/).

#### Using DistSQL to implement Hint has the following problems:
1. To execute a query, the user needs to send `Hint DistSQL` and `DML SQL` separately, which is not friendly to use;
2. After using `Hint DistSQL`, the user needs to send an additional `CLEAR` statement;
3. When Hint is enabled, Proxy needs to make a special judgment on the connected thread, which increases the complexity; ([link](https://github.com/apache/shardingsphere/blob/efe9ee21266d6507dee4639eb0c19a6d98992325/proxy/frontend/core/src/main/java/org/apache/shardingsphere/proxy/frontend/state/impl/OKProxyState.java#L61))

Now, SQL Hint can perfectly solve the above problems, so I suggest to start the work of removing Hint DistSQL, with the following tasks:

- [x] Remove `HintRALStatementExecutor` and implementation
- [x] Remove `HintRALBackendHandler`
- [x] Remove `HintRALStatement` and subclasses
- [x] Remove Hint DistSQL documentation
- [ ] Remove the judgment related to Hint in `OKProxyState`
- [ ] Remove the configuration item: `proxy-hint-enabled`
- [x] Remove related test cases
- [x] Remove HintManagerHolder

### PR DESCRIPTION
**Pull request by anmui**
Fixes #23953.

Changes proposed in this pull request:
- [x]  Remove HintRALStatementExecutor and implementation
- [x]  Remove HintRALBackendHandler
- [x]  Remove HintRALStatement and subclasses
- [x]  Remove Hint DistSQL documentation
- [x]  Remove the judgment related to Hint in OKProxyState
- [ ]  Remove the configuration item: proxy-hint-enabled
- [x]  Remove related test cases
- [x]  Remove HintManagerHolder

---

Before committing this PR, I'm sure that I have checked the following options:
- [ ] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [ ] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [ ] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: features/sharding/core/src/test/java/org/apache/shardingsphere/sharding/route/engine/type/standard/ShardingStandardRoutingEngineTest.java [hunk 1]
```diff
@@ -45,7 +44,6 @@ public final class ShardingStandardRoutingEngineTest extends AbstractRoutingEngi
     
     @After
     public void tearDown() {
```
Comment (RaigorJiang) [inline features/sharding/core/src/test/java/org/apache/shardingsphere/sharding/route/engine/type/standard/ShardingStandardRoutingEngineTest.java:46]: Can this method `tearDown()` be removed?

Comment (anmui) [reply features/sharding/core/src/test/java/org/apache/shardingsphere/sharding/route/engine/type/standard/ShardingStandardRoutingEngineTest.java:46]: ok,and i removed HintManager & HintManagerHolder,is it OK？

Comment (anmui) [reply features/sharding/core/src/test/java/org/apache/shardingsphere/sharding/route/engine/type/standard/ShardingStandardRoutingEngineTest.java:46]: > Can this method `tearDown()` be removed?
ok,and i removed HintManager & HintManagerHolder,is it OK？

Comment (anmui) [reply features/sharding/core/src/test/java/org/apache/shardingsphere/sharding/route/engine/type/standard/ShardingStandardRoutingEngineTest.java:46]: > Can this method `tearDown()` be removed?

ok,and i removed HintManager & HintManagerHolder,is it OK？

Comment (RaigorJiang) [reply features/sharding/core/src/test/java/org/apache/shardingsphere/sharding/route/engine/type/standard/ShardingStandardRoutingEngineTest.java:46]: @anmui Yes, they should be removed.

**Comment by RaigorJiang** [comment]
Hi @anmui , Please fix CI, you may need to remove Hint test cases first.

**Comment by anmui** [comment]
> Hi @anmui , Please fix CI, you may need to remove Hint test cases first.

OK,thanks

**Comment by RaigorJiang** [comment]
@anmui 
Sorry, I noticed that `HintManager` still needs to be used in the Java API, so it can't be removed yet.
Please update and restore HintManager, thank you!

**Comment by anmui** [comment]
> @anmui Sorry, I noticed that `HintManager` still needs to be used in the Java API, so it can't be removed yet. Please update and restore HintManager, thank you!

OK,Can I ask where it is still used? 😆

**Comment by RaigorJiang** [comment]
> OK,Can I ask where it is still used? 😆

Please refer to:
https://shardingsphere.apache.org/document/current/en/user-manual/shardingsphere-jdbc/special-api/sharding/hint/

**Comment by anmui** [comment]
> > OK,Can I ask where it is still used? 😆
> 
> Please refer to: https://shardingsphere.apache.org/document/current/en/user-manual/shardingsphere-jdbc/special-api/sharding/hint/

Get it,thank you!

**Comment by RaigorJiang** [comment]
Hi @anmui , please fix CI.

**Comment by anmui** [comment]
@RaigorJiang hi,can it test again,It seems that the port conflicts,i am not sure 🥲

**Comment by RaigorJiang** [comment]
@anmui  There is a file conflict now.

**Comment by anmui** [comment]
It seems that the E2ETestEnvironment cannot be initialized...

**Comment by RaigorJiang** [comment]
I found that there is an issue working on this: #24254

**Comment by anmui** [comment]
@RaigorJiang hi,i want to make a test🙋

**Comment by RaigorJiang** [comment]
@anmui 
You can compile and test locally before pushing:
> ./mvnw clean install -Djacoco.skip=true -Dmaven.javadoc.skip=true -Drat.skip=true

**Comment by anmui** [comment]
> @anmui You can compile and test locally before pushing:
> 
> > ./mvnw clean install -Djacoco.skip=true -Dmaven.javadoc.skip=true -Drat.skip=true

OK,it BUILD SUCCESS in my test😢

**Comment by anmui** [comment]
@RaigorJiang 
I seem to know why it always reports that connect close is wrong 😢

If **_proxy-hint-enabled=true_**, the thread processing model of Proxy-OKProxyState is changed from IO multiplexing to an independent thread for each request

The **_proxy-hint-enabled=true_** configured in the test environment, so it will establish an independent thread for each request. If the parameter is removed, it will become a reused connection. If the reused connection is closed, other requests cannot be executed

       System.out.println(requireOccupyThreadForConnection(connectionSession));  //false
       System.out.println(connectionSession.getTransactionStatus().getTransactionType()); //false
        if (requireOccupyThreadForConnection(connectionSession)) {
            return ConnectionThreadExecutorGroup.getInstance().get(connectionSession.getConnectionId());
        }
        System.out.println("isPreferNettyEventLoop:" + isPreferNettyEventLoop()); //false
        if (isPreferNettyEventLoop()) {
            return context.executor();
        }
        System.out.println("isRequiredSameThreadForConnection:" + databaseProtocolFrontendEngine.getFrontendContext().isRequiredSameThreadForConnection(message)); //false
        if (databaseProtocolFrontendEngine.getFrontendContext().isRequiredSameThreadForConnection(message)) {
            return ConnectionThreadExecutorGroup.getInstance().get(connectionSession.getConnectionId());
        }
        // return UserExecutorGroup.getInstance().getExecutorService();
        return ConnectionThreadExecutorGroup.getInstance().get(connectionSession.getConnectionId());`

👆Change it can success run

So how do I deal with it? 
Do I need to remove the **_proxy-hint-enabled=true_**

**Comment by anmui** [comment]
@RaigorJiang 
截屏2023-02-21 21 00 01
👆the error message
截屏2023-02-21 21 02 17
👆Transaction statics

**Comment by RaigorJiang** [comment]
@anmui 
Yes, `proxy-hint-enabled` should be removed, it's just for Hint DistSQL.

**Comment by anmui** [comment]
@anmui Yes, `proxy-hint-enabled` should be removed, it's just for Hint DistSQL.
But after `proxy-hint-enabled` is deleted, the test transaction type is local and the connection will be reused. The E2E test will fails. 
I solve this problem by establishing a connection for each request even if the transaction type is local in the proxy. 
Can we solve this problem in this way? 😂
I guess it can be solved by changing the configuration of the test environment..🤔

     transaction:
        defaultType: XA
        providerType: Atomikos

**Comment by RaigorJiang** [comment]
> @anmui Yes, `proxy-hint-enabled` should be removed, it's just for Hint DistSQL. But after `proxy-hint-enabled` is deleted, the test transaction type is local and the connection will be reused. The E2E test will fails. 😭 I solve this problem by establishing a connection for each request even if the transaction type is local in the proxy. Can we solve this problem in this way? 😂 I guess it can be solved by changing the configuration of the test environment..🤔
> 
> ```
>  transaction:
>     defaultType: XA
>     providerType: Atomikos
> ```

Hi @taojintianxia , can you take a look?  Thank you!

**Comment by taojintianxia** [comment]
> > @anmui Yes, `proxy-hint-enabled` should be removed, it's just for Hint DistSQL. But after `proxy-hint-enabled` is deleted, the test transaction type is local and the connection will be reused. The E2E test will fails. 😭 I solve this problem by establishing a connection for each request even if the transaction type is local in the proxy. Can we solve this problem in this way? 😂 I guess it can be solved by changing the configuration of the test environment..🤔
> > ```
> >  transaction:
> >     defaultType: XA
> >     providerType: Atomikos
> > ```
> 
> Hi @taojintianxia , can you take a look? Thank you!

this seems not a environment issue, I suggest to take a look on the logic of the corresponding code.

**Comment by anmui** [comment]
> 

      private boolean requireOccupyThreadForConnection(final ConnectionSession connectionSession) {
              return ProxyContext.getInstance().getContextManager().getMetaDataContexts().getMetaData().getProps().<Boolean>getValue(ConfigurationPropertyKey.PROXY_HINT_ENABLED)
                      || TransactionType.isDistributedTransaction(connectionSession.getTransactionStatus().getTransactionType());
      }

This is the previous code,if `PROXY_HINT_ENABLED` is true,this will return true

the ExecutorService will  build in this way

    ConnectionThreadExecutorGroup.getInstance().get(connectionSession.getConnectionId());  // ⚠️Prototype

and we need to delete  `PROXY_HINT_ENABLED`,the code is

    private boolean requireOccupyThreadForConnection(final ConnectionSession connectionSession) {
              return  TransactionType.isDistributedTransaction(connectionSession.getTransactionStatus().getTransactionType());
      }
when `TransactionType` is local,it will return false

the ExecutorService will build in this way

    UserExecutorGroup.getInstance().getExecutorService();  // ⚠️Singleton

when E2E test,it will failed,because connect close 😢

**Comment by RaigorJiang** [comment]
> when E2E test,it will failed,because connect close 😢
> 
> do you understand what I mean😭

Thanks for your feedback, I will test and verify.

**Comment by RaigorJiang** [comment]
Hi @anmui 
I verified in #24298. As you said, removing `proxy-hint-enabled` brought the "connection closed" phenomenon.
The root cause of the problem still needs to be investigated. 
I propose to keep the `proxy-hint-enabled` in this PR for the time being, and delete it after the problem is solved. 
Do you think it is ok?

**Comment by anmui** [comment]
> I propose to keep the `proxy-hint-enabled` in this PR for the time being, and delete it after the problem is solved.
> Do you think it is ok?

OK,I will restore. and I want to know if Remove Hint DistSQL document is deleted [here](https://shardingsphere.apache.org/document/current/en/user-manual/shardingsphere-proxy/distsql/syntax/ral/hint/)？

**Comment by RaigorJiang** [comment]
> > I propose to keep the `proxy-hint-enabled` in this PR for the time being, and delete it after the problem is solved.
> > Do you think it is ok?
> 
> OK,I will restore. and I want to know if Remove Hint DistSQL document is deleted [here](https://shardingsphere.apache.org/document/current/en/user-manual/shardingsphere-proxy/distsql/syntax/ral/hint/)？

Yes, we only need to remove the corresponding directory [here](https://github.com/apache/shardingsphere/tree/master/docs/document/content/user-manual/shardingsphere-proxy/distsql/syntax/ral) , the official website will automatically refresh.

File: proxy/bootstrap/src/main/resources/conf/server.yaml [hunk 2]
```diff
@@ -86,3 +86,18 @@
 #  proxy-default-port: 3307 # Proxy default port.
 #  proxy-netty-backlog: 1024 # Proxy netty backlog.
 #  cdc-server-port: 33071 # CDC server port
+
+authority:
+  users:
```
Comment (RaigorJiang) [inline proxy/bootstrap/src/main/resources/conf/server.yaml:91]: Hi @anmui , please do not push the config file.

**Comment by RaigorJiang** [comment]
@anmui 
A great work, thank you!

**Pull request merged**